flycheck.el

```diff
@@ -2792,6 +2837,62 @@ because some users override that function, as described in URL
   (flycheck-clear-idle-change-timer)
   (flycheck-buffer-automatically 'idle-change))
 
+(defvar-local flycheck-idle-buffer-switch-timer nil
+  "Timer to mark the idle time since switching to a new buffer.")
+
+(defvar-local flycheck--force-checking-intermediate-buffers nil
+  "If non-nil, don't cancel syntax checks for intermediate buffers.
+Ignore `flycheck-buffer-switch-check-intermediate-buffers' in
+that case.")
+
+(defun flycheck-clear-idle-buffer-switch-timer ()
+  "Clear the idle buffer switch timer."
+  (when flycheck-idle-buffer-switch-timer
+    (cancel-timer flycheck-idle-buffer-switch-timer)
+    (setq flycheck-idle-buffer-switch-timer nil)))
+
+(defvar flycheck--last-buffer (current-buffer)
+  "The current buffer or the buffer that was previously current.
+This is usually equal to the current buffer, unless the user just
+switched buffers.  After a buffer switch, it is the previous
+buffer.")
+
+(defun flycheck-handle-buffer-switch ()
+  "Handle a possible switch to another buffer.
+
+If a buffer switch actually happened, schedule a syntax check."
+  (unless (equal flycheck--last-buffer (current-buffer))
+    (setq flycheck--last-buffer (current-buffer))
+    (when flycheck-mode
+      ;; Reschedule any check that was going to be run in the current
+      ;; buffer due to a buffer switch.
+      (flycheck-clear-idle-buffer-switch-timer)
+      ;; If we can guarantee that this timer is actually going to
+      ;; result in a syntax check, we can cancel any other
+      ;; pre-existing timers that would cause syntax checks.
+      (when (memq 'idle-buffer-switch flycheck-check-syntax-automatically)
+        ;; Make sure that this timer can't get canceled by the user
+        ;; switching to another buffer.
+        (setq flycheck--force-checking-intermediate-buffers t)
+        (flycheck-clear-idle-change-timer))
+      (setq flycheck-idle-buffer-switch-timer
+            (run-at-time flycheck-idle-buffer-switch-delay nil
+                         #'flycheck-handle-idle-buffer-switch
+                         (current-buffer))))))
+
+(defun flycheck-handle-idle-buffer-switch (buffer)
+  "Handle an expired idle timer in BUFFER since the last buffer switch."
+  (when (buffer-live-p buffer)
+    (with-current-buffer buffer
+      (flycheck-clear-idle-buffer-switch-timer)
+      (when (or flycheck-buffer-switch-check-intermediate-buffers
```

Comment:
I'm not sure I understand this condition.  If `-check-intermediate-buffers` is non-nil, then all idle-buffer-switch should trigger a check?

I don't get the `(equal buffer (current-buffer))` however.  Isn't that always the case since we are in `with-current-buffer buffer`?

Reply:
Yes, if `flycheck-buffer-switch-check-intermediate-buffers` is non-nil, then `flycheck-idle-buffer-switch-delay` seconds after you switch to a buffer, Flycheck will run a syntax check, regardless of whether the user is still in that buffer.

The `(equal buffer (current-buffer))` was a mistake; I didn't realize that `(current-buffer)` was inside a `(with-current-buffer ...)`. That has now been fixed.